Src/ReadExternalPressure.c

```diff
@@ -3,18 +3,60 @@
 #include "cmsis_os.h"
 
 #include "ReadExternalPressure.h"
-
 #include "Data.h"
 
-static int READ_EXTERNAL_PRESSURE_PERIOD = 1000;
+static const int READ_EXTERNAL_PRESSURE_PERIOD = 100;
+
+static const int CMD_SIZE = 1;
+static const int CMD_TIMEOUT = 150;
+static const uint8_t ADC_512_CONV_CMD = 0x42;
+static const uint8_t ADC_READ_CMD = 0x00;
+static const uint8_t RESET_CMD = 0x1E;
+
+static uint8_t dataIn;
 
 void readExternalPressureTask(void const* arg)
 {
     ExternalPressureData* data = (ExternalPressureData*) arg;
     uint32_t prevWakeTime = osKernelSysTick();
 
+    HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_RESET);
+    HAL_SPI_Transmit(&hspi2, &RESET_CMD, CMD_SIZE, CMD_TIMEOUT);
+    osDelay(3);   // 3 ms after reset
```

Comment:
In general, comment should say why not what. This comment just says what, but doesn't explain why this delay is necessary.